stakingModel/test/BasicCalculation.test.js

```diff
@@ -0,0 +1,130 @@
+const InterestDistributionMock = artifacts.require("InterestDistributionMock");
+
+const BN = web3.utils.BN;
+
+
+contract("InterestDistribution - Basic calculations", ([user1]) => {
+  
+
+  let interestDistribution;
+  before(async () => {
+    interestDistribution = await InterestDistributionMock.new();
+    
+  });
+
+  it("Should return correct Accumulated Yield Per Token", async () => {
+
+    /**
+      * Formula:
+      * AccumulatedYieldPerToken = AccumulatedYieldPerToken(P) + (Daily Interest/GrandTotalStaked)
+      * consider :
+      * AccumulatedYieldPerToken(P) = 0
+      * Daily Interest = 10 x 1e2
+      * GrandTotalStaked = 100 x 1e18
+      * output = 10
+      */
+
+      let accumulatedYieldPerToken = await interestDistribution.getAccumulatedYieldPerToken(10*1e2, web3.utils.toWei("100", "ether"));
+      
+      expect(accumulatedYieldPerToken.toString()).to.be.equal("10");
+  });
+
+  it("Should return correct Avg Yield Rate Per Token", async () => {
+
+    /**
+      * Formula:
+      * AvgYieldRatePerToken = [(AvgYieldRatePerToken(P) x TotalStaked) + (AccumulatedYieldPerToken(P) x Staking x (1-%Donation))]/TotalStaked
+      * consider :
+      * AvgYieldRatePerToken(P) = 0
+      * AccumulatedYieldPerToken(P) = 10
+      * Staking = 10 x 1e18
+      * Donation% = 30%
+      * TotalStaked = 20 x 1e18
+      * output = 3
+      */
+
+      let avgYieldRatePerToken = await interestDistribution.getAvgYieldRatePerToken(10, web3.utils.toWei("10", "ether"), 30, web3.utils.toWei("20", "ether"));
+      
+      expect(avgYieldRatePerToken.toString()).to.be.equal("3");
+  });
+
+  it("Set Accumulated Yield Per Token and Avg Yield Rate Per Token", async () => {
+
+
+    await interestDistribution
+        .addAccumulatedYieldPerToken(10)
+        .catch(e => e);
+
+    await interestDistribution
+        .stakeCalculation(user1, web3.utils.toWei("10", "ether"), 0, 70*1e2, web3.utils.toWei("1000", "ether"))
+        .catch(e => e);
+
+    let yieldData = await interestDistribution.getYieldData(user1);
+    
+    expect(yieldData[0].toString()).to.be.equal("10");
```

Comment:
in all expected results where you use hard coded numbers explain in words or with formula why do you expect this number

Reply:
Done.